src/python/pants/engine/internals/scheduler.py

```diff
@@ -155,19 +156,47 @@ def __init__(
             interactive_process_result=InteractiveProcessResult,
             engine_aware_parameter=EngineAwareParameter,
         )
+        remoting_options = PyRemotingOptions(
+            execution_enable=execution_options.remote_execution,
+            store_address=execution_options.remote_store_address,
+            execution_address=execution_options.remote_execution_address,
+            execution_process_cache_namespace=execution_options.process_execution_cache_namespace,
+            instance_name=execution_options.remote_instance_name,
+            root_ca_certs_path=execution_options.remote_ca_certs_path,
+            store_headers=tuple(execution_options.remote_store_headers.items()),
+            store_chunk_bytes=execution_options.remote_store_chunk_bytes,
+            store_chunk_upload_timeout=execution_options.remote_store_chunk_upload_timeout_seconds,
+            store_rpc_retries=execution_options.remote_store_rpc_retries,
+            cache_eager_fetch=execution_options.remote_cache_eager_fetch,
+            execution_extra_platform_properties=tuple(
+                tuple(pair.split("=", 1))
+                for pair in execution_options.remote_execution_extra_platform_properties
+            ),
+            execution_headers=tuple(execution_options.remote_execution_headers.items()),
+            execution_overall_deadline_secs=execution_options.remote_execution_overall_deadline_secs,
+        )
+        exec_stategy_opts = PyExecutionStrategyOptions(
+            local_parallelism=execution_options.process_execution_local_parallelism,
+            remote_parallelism=execution_options.process_execution_remote_parallelism,
+            cleanup_local_dirs=execution_options.process_execution_cleanup_local_dirs,
+            use_local_cache=execution_options.process_execution_use_local_cache,
+            remote_cache_read=execution_options.remote_cache_read,
+            remote_cache_write=execution_options.remote_cache_write,
+        )
 
-        self._scheduler = native.new_scheduler(
-            tasks=tasks,
-            build_root=build_root,
-            local_store_dir=local_store_dir,
-            local_execution_root_dir=local_execution_root_dir,
-            named_caches_dir=named_caches_dir,
-            ca_certs_path=ca_certs_path,
-            ignore_patterns=ignore_patterns,
-            use_gitignore=use_gitignore,
-            executor=executor,
-            execution_options=execution_options,
-            types=types,
+        self._py_scheduler = native_engine.scheduler_create(
```

Comment:
Note the rename from `._scheduler` to `._py_scheduler`, which clears up one of my biggest confusions: what did `Scheduler._scheduler` mean?